src/SoundOptionsModal.test.js

```diff
@@ -0,0 +1,114 @@
+            wrappingComponent: IntlProvider,
+            wrappingComponentProps: {
+                locale: 'en',
+                defaultLocale: 'en',
+                messages: messages.en
+            }
+        }
+    );
+
+    return {
+        wrapper,
+        mockOnCancel,
+        mockOnSave
+    };
+}
+
+function getAudioToggleSwitches(wrapper) {
+    return wrapper.find(ToggleSwitch);
+}
+
+function getCancelButton(wrapper) {
+    return wrapper.find('.ModalWithFooter__secondaryButton');
+}
+
+function getSaveButton(wrapper) {
+    return wrapper.find('.ModalWithFooter__primaryButton');
+}
+
+describe('When rendering audio toggles', () => {
+    test('Three toggle switches should render in following order: "All Sounds", "Musical Sounds" and "Audio Announcements"', () => {
+        expect.assertions(7);
+        const { wrapper } = createMountSoundOptionsModal();
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(true);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(true);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(true);
+    });
+    test('Changing audioEnabled, announcementsEnabled and sonificationEnabled properties to false should turn off corresponding toggle siwtches', () => {
+        expect.assertions(7);
+        const { wrapper } = createMountSoundOptionsModal({ audioEnabled: false, announcementsEnabled: false, sonificationEnabled: false });
+        const audioToggleSwitches = getAudioToggleSwitches(wrapper);
+        expect(audioToggleSwitches.length).toBe(3);
+        expect(audioToggleSwitches.get(0).props.id).toBe('sound-options-allsounds');
+        expect(audioToggleSwitches.get(0).props.value).toBe(false);
+        expect(audioToggleSwitches.get(1).props.id).toBe('sound-options-musicalSounds');
+        expect(audioToggleSwitches.get(1).props.value).toBe(false);
+        expect(audioToggleSwitches.get(2).props.id).toBe('sound-options-announcements');
+        expect(audioToggleSwitches.get(2).props.value).toBe(false);
+    })
+});
+
+describe('When interacting with footer buttons', () => {
+    test('Pressing cancel button should close the modal without changing toggle states', () => {
+        expect.assertions(2);
+        const { wrapper, mockOnCancel } = createMountSoundOptionsModal();
+        const cancelButton = getCancelButton(wrapper);
+        cancelButton.at(0).simulate('click');
+        expect(mockOnCancel.mock.calls.length).toBe(1);
+        expect(mockOnCancel.mock.calls[0][0]).toBe(undefined);
+    });
+    test('Pressing save button should update the toggle states', () => {
```

Comment:
Again, this doesn't seem like what you're actually testing.  You're confirming that the save button is wired up, but you're not actually making a change or confirming that it's passed along on a save.  You can either change the description or (my preference) actually test what you describe.

I used the same pattern in the keyboard input modal work, just testing the functions, but in later work (on the actions simplification menu), it seemed better to actually test the change (or lack of change) as well.

Reply:
Good suggestions, I updated both cancel and save button to check its state changes.